src/state/manager_wrapper.rs

```diff
@@ -111,9 +123,12 @@ impl ManagerWrapper {
             .collect();
         let old_account_roots: Vec<Fr> = buffered_txs.iter().map(|tx| tx.root_before).collect();
         let new_account_roots: Vec<Fr> = buffered_txs.iter().map(|tx| tx.root_after).collect();
+        //calc tx-pubdata's hash
+        buffered_txs.iter().for_each(|tx| tx.encode_pubdata(encoder).unwrap());
```

Comment:
This encode for each `RawTx`, but we want to encode according to their types (`TransferTx`, `WithdrawTx`...)

Reply:
Currently the encoding in circuit is somewhat 'uniform': i.e. it just encode some of the fields from the encodedTxs input signals. RawTx::encode_pubdata and these codes are just the Rust mapping of what we have done inside circuits.

I guess it would be hard enough to include tx type in the tx pubdata and here should we insist on encoding txs from the typed form (TransferTx, WithdrawTx ...) rather than the encoded one?